test/e2e/pull_chunked_test.go

```diff
@@ -0,0 +1,393 @@
+//go:build linux || freebsd
+
+package integration
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"slices"
+	"strings"
+
+	. "github.com/containers/podman/v5/test/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+	"github.com/opencontainers/go-digest"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func pullChunkedTests() { // included in pull_test.go
```

Comment:
All of this is rather tedious.

Plausibly we could pre-build all the test images and ship them in the CI images… but we still need to `podman system reset` for each test case, and the pulls must run from a real registry (not from `oci:` nor using `podman load`), so I’m not sure that would be any better.

It’s quite likely there is some infrastructure that could shorten these tests; I’d welcome pointers.